### PR TITLE
tables: adding scheduled action to windows scheduled tasks table

### PR DESCRIPTION
**Pull request by muffins**
This addresses #3254

### CODE REVIEW

File: osquery/tables/system/windows/scheduled_tasks.cpp [hunk 1]
```diff
@@ -136,6 +136,53 @@ void enumerateTasksForFolder(std::string path, QueryData& results) {
     LocalFileTimeToFileTime(&ft, &locFt);
     r["next_run_time"] = INTEGER(filetimeToUnixtime(locFt));
 
+    ITaskDefinition* taskDef = nullptr;
+    pRegisteredTask->get_Definition(&taskDef);
+
+    IActionCollection* tActionCollection = nullptr;
+    taskDef->get_Actions(&tActionCollection);
+    long actionCount = 0;
+    tActionCollection->get_Count(&actionCount);
```
Comment (obelisk) [inline osquery/tables/system/windows/scheduled_tasks.cpp]: Can this be a nullptr?

File: osquery/tables/system/windows/scheduled_tasks.cpp [hunk 2]
```diff
@@ -136,6 +136,53 @@ void enumerateTasksForFolder(std::string path, QueryData& results) {
...
+    std::vector<std::string> actions;
+
+    // Task collections are 1-indexed
+    for (long j = 1; j <= actionCount; j++) {
```
Comment (obelisk) [inline osquery/tables/system/windows/scheduled_tasks.cpp]: Nit: Could be auto or a size_t

File: osquery/tables/system/windows/scheduled_tasks.cpp [hunk 3]
```diff
@@ -136,6 +136,53 @@ void enumerateTasksForFolder(std::string path, QueryData& results) {
...
+      actions.push_back(wstringToString(full.c_str()));
+      act->Release();
+    }
+    if (tActionCollection != nullptr) {
+      tActionCollection->Release();
+    }
+    if (taskDef != nullptr) {
+      taskDef->Release();
+    }
+    r["action"] = actions.size() > 0 ? osquery::join(actions, ",") : "";
```
Comment (obelisk) [inline osquery/tables/system/windows/scheduled_tasks.cpp]: Prefer empty()

**Comment by osqueryer** [comment]
:no_good: The commit 612853475144ca313737e68be2486d86e4f34a5a (Job results: [2193](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2193/)) failed the code audit or documentation test.

**Comment by muffins** [comment]
@obelisk will double check, i don't think we can early exit on these as we're enumerating loops, and we need to release resources at the end. Also this is _not_ ready to merge as I broke something :)

**Comment by osqueryer** [comment]
:no_good: The commit 62319ddb6f48024ca47c733af2324a9316d265b8 (Job results: [2195](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2195/)) failed the code audit or documentation test.

**Pull request merged**